File: Lista3/Lista3_11800910/exec3/Circulo.java
import java.lang.Math.*;

public class Circulo extends Forma {

    private double raio; //raio

    /**
     * Construtor do circulo
     * @param raio
     * @param cheio
     * @param cor
     */
    public Circulo(double raio, boolean cheio, String cor){
        super(cheio, cor);
        this.raio = raio;
    }

    /**
     * Sobreescreve a area
     */
    @Override
    public double perimetro(){
        return 2 * Math.PI * raio;
    }

    /**
     * Sobreescreve o perimetro
     */
    @Override
    public double area(){
        return Math.PI * raio * raio;
    }

    /**
     * Representação do circulo como string
     */
    @Override
    public String toString(){

        boolean cheio = getCheio();
        String s = "\nForma Geométrica: CIRCULO\n";
        s += "Raio: " + raio + "\n";
        s += "Area: " + area() + "\n";
        s += "Perímetro: " + perimetro() + "\n";
        s += "Colorido: " + cheio + "\n";
        if(cheio){ 
            s += "Cor: " + getCor() + "\n"; 
        }
        return s;
    }

}

File: Lista4/exec3/circulo.py
from forma import *
import math

class circulo(forma):

    def __init__(self, raio: float, cheio: bool, cor: str) -> None:
        super().__init__(0, 0, cheio, cor)
        self.raio = raio

    def area(self):
        return float(math.pi) * self.raio * self.raio

    def perimetro(self):
        return 2 * math.pi * self.raio

    def __str__(self) -> str:
        self.cheio = self.getCheio()
        s = '\nForma Geométrica: CIRCULO\n'
        s += "Lados: " + str(self.raio) + "\n"
        s += "Area: " + str(self.area()) + "\n"
        s += "Perímetro: " + str(self.perimetro()) + "\n"
        s += "Colorido: " + str(self.cheio) + "\n"
        if(self.cheio):
            s += "Cor: " + self.getCor() + "\n"
        
        return s

